src/WebJobs.Extensions.OpenAI/Embeddings/EmbeddingsStoreConverter.cs

```diff
@@ -49,8 +49,9 @@ public Task<IAsyncCollector<SearchableDocument>> ConvertAsync(EmbeddingsStoreAtt
     internal SearchableDocument ToSearchableDocument(string? json)
     {
         this.logger.LogDebug("Creating searchable document from JSON string: {Text}", json);
-        SearchableDocument document = JsonSerializer.Deserialize<SearchableDocument>(json, options);
-        return document ?? throw new ArgumentException("Invalid search request.");
+        SearchableDocument document = JsonSerializer.Deserialize<SearchableDocument>(json ?? throw new ArgumentNullException(nameof(json)), options)
```

Comment:
nit: I suggest doing the `json` null check earlier in the method - i.e., `ArgumentNullException.ThrowIfNull(json)`, which is what you've been doing in other places. That will help reduce line lengths.

Reply:
ArgumentNullException.ThrowIfNull is not available for .Net Standard 2.1, I moved the null check to top with if clause.